`typing_extensions.NewType` annotations are not recognized on < 3.10 (regression in 1.4.0)

**Bug Report**

Mypy 1.4.0 introduced a regression where `typing_extensions.NewType` annotations are not correctly recognized when `--python-version` < 3.10.

this bisects to https://github.com/python/typeshed/commit/a6d0aaf855061647dd4166db9b4b8825ad941c30 via https://github.com/python/typeshed/pull/10205 via https://github.com/python/mypy/pull/15334. however this does not appear to be a typeshed issue; it's just that the typeshed change exposed a previously-hidden Mypy issue.

**To Reproduce**

```python
# run with --python-version=3.9 or below, using mypy d11db28 or 1.4 or later

from typing_extensions import NewType

N = NewType("N", int)

x: N  # error: Variable "eg_newtype.N" is not valid as a type  [valid-type]
```

**Expected Behavior**

Mypy passes this example.

**Actual Behavior**

Mypy fails this example; see above.

**Your Environment**

<!-- Include as many relevant details about the environment you experienced the bug in -->

- Mypy version used: 1.6.1 (compiled) and current master (uncompiled)
- Mypy command-line flags: `--python-version=3.9`
- Mypy configuration options from `mypy.ini` (and other config files): none (i.e. using vendored typeshed)
- Python version used: 3.12

<!-- You can freely edit this text, please remove all the lines you believe are unnecessary. -->

--- a/mypy/semanal_newtype.py
+++ b/mypy/semanal_newtype.py
@@ -147,7 +147,7 @@ class NewTypeAnalyzer:
             and isinstance(s.lvalues[0], NameExpr)
             and isinstance(s.rvalue, CallExpr)
             and isinstance(s.rvalue.callee, RefExpr)
-            and s.rvalue.callee.fullname == "typing.NewType"
+            and (s.rvalue.callee.fullname in ("typing.NewType", "typing_extensions.NewType"))
         ):
             name = s.lvalues[0].name
 

--- a/test-data/unit/check-newtype.test
+++ b/test-data/unit/check-newtype.test
@@ -379,3 +379,10 @@ N = NewType('N', XXX)  # E: Argument 2 to NewType(...) must be subclassable (got
                        # E: Name "XXX" is not defined
 x: List[Union[N, int]]
 [builtins fixtures/list.pyi]
+
+[case testTypingExtensionsNewType]
+# flags: --python-version 3.7
+from typing_extensions import NewType
+N = NewType("N", int)
+x: N
+[builtins fixtures/tuple.pyi]

